torch/ao/quantization/fx/_model_report/model_report.py

```diff
@@ -436,7 +448,76 @@ def generate_qconfig_mapping(self) -> QConfigMapping:
 
         Returns a QConfigMapping for the quantization configuration
         """
-        pass
+        # first check if user has at least done report generation
+        if len(self._generated_reports) == 0:
+            raise Exception("Unable to generate QConfigMapping without first generating reports and suggestions")
+
+        # we can use the report suggestions from each detector to compose our mappings
+        # get the reformatted reports
+        reformatted: Dict[str, Dict[str, Any]] = self._reformat_reports_for_visualizer()
+
+        # now we create a high level mapping with a default qconfig
+        return_mapping = QConfigMapping()
+        return_mapping.set_global(default_qconfig)
+
+        # get the set of modules that are support per channel
+        current_backend = torch.backends.quantized.engine
```

Comment:
also I feel it might make more sense to ask user for the backend, the quantized engine when we call this function may not be the same as the quantized engine during actual inference